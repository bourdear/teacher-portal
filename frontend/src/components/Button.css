.add-student {
  border-radius: 15px;
  border: solid 2px rgb(212, 212, 212);
  background: white;
  padding: 5px;
  width: 150px;
  font-weight: 700;
  transition: 0.3s;
}

.add-student:hover {
  background: 2px rgb(212, 212, 212);
  cursor: pointer;
}