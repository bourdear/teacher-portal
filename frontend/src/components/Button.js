import React from 'react'
import './Button.css'

const Button = ({ value, name, handleClick }) => (
  <input type='button'
   value={value} 
   name={name}
   onClick={handleClick}
   className='add-student'/>
)

export default Button;