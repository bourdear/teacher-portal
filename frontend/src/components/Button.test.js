import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import Button from './Button';

configure({adapter: new Adapter()})
const mockFunction = jest.fn()
mockFunction.mockReturnValue('clicked')

let wrapper;
const buttonData = {
  value: 'click here',
  name: 'btn1',
  handleClick: mockFunction
}

test('component renders', () => {
  const component = renderer.create(
    <Button value={buttonData.value} name={buttonData.name} handleClick={buttonData.handleClick}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('wrapper exists', () => {
  wrapper = shallow(<Button value={buttonData.value} name={buttonData.name} handleClick={buttonData.handleClick} />)
  expect(wrapper.exists()).toBeTruthy()
})

test('props are correct', () => {
  expect(wrapper.props().value).toEqual('click here')
  expect(wrapper.props().name).toEqual('btn1')
})

test('button has been clicked', () => {
  wrapper.simulate(('click'))
  expect(buttonData.handleClick.mock.calls.length).toEqual(1)
})



