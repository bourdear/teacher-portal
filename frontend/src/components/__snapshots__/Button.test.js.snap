// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<input
  className="add-student"
  name="btn1"
  onClick={[MockFunction]}
  type="button"
  value="click here"
/>
`;
